gems/pending/spec/spec_helper.rb

```diff
@@ -32,3 +36,30 @@
   config.backtrace_exclusion_patterns << %r{/lib\d*/ruby/[0-9]}
   config.backtrace_exclusion_patterns << %r{/gems/[0-9][^/]+/gems/}
 end
+
+#
+# So tests can clear class-level caches between examples.
+# TODO: Add this to the azure-armrest gem.
+#
+class Azure::Armrest::ArmrestService
+  def self.clear_caches
+    @@providers_hash = {}
+    @@tokens         = {}
+    @@subscriptions  = {}
+  end
+end
```

Comment:
There isn't a good way to load it on a case by case basis (RSpec provides just an `include` option)...until about a week ago when a new hook was merged :) So will have to do that later, it'll be just a normal `config.include` here for now.

I'd expect any helpers to live in `gems/pending/spec/support/`